From 28688a2c33f7691e0040678d618f18330952de08 Mon Sep 17 00:00:00 2001
From: Mahesh Kumar P <mahesh.kumar.p@intel.com>
Date: Mon, 9 Feb 2015 17:17:36 +0530
Subject: [PATCH] x86, hpet: Add quirk to disable HPET for Cherrytrail platform

HPET on CHT ,like on VLV,  has accuracy problem to be used as
reliable clocksource/clockevent, So add a early quirk to disable it.
Also,HPET should only be used for clock calibration which should be done
before any Cstates or RTD3.

Issue: GMIN-3635

Change-Id: I4d18439de6024752622c8259a3e75811a8b34bf5
Signed-off-by: Kumar P Mahesh <mahesh.kumar.p@intel.com>
---
 arch/x86/kernel/early-quirks.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/arch/x86/kernel/early-quirks.c b/arch/x86/kernel/early-quirks.c
index 2e1a685..aaa22c6 100644
--- a/arch/x86/kernel/early-quirks.c
+++ b/arch/x86/kernel/early-quirks.c
@@ -604,11 +604,14 @@ static struct chipset early_qrk[] __initdata = {
 	{ PCI_VENDOR_ID_INTEL, PCI_ANY_ID, PCI_CLASS_DISPLAY_VGA, PCI_ANY_ID,
 	  QFLAG_APPLY_ONCE, intel_graphics_stolen },
 	/*
-	 * HPET on current version of Baytrail platform has accuracy
+	 * HPET on current version of Baytrail/Cherrytrail platform has accuracy
 	 * problems, disable it for now:
 	 */
 	{ PCI_VENDOR_ID_INTEL, 0x0f00,
 		PCI_CLASS_BRIDGE_HOST, PCI_ANY_ID, 0, force_disable_hpet},
+	{ PCI_VENDOR_ID_INTEL, 0x2280,
+		PCI_CLASS_BRIDGE_HOST, PCI_ANY_ID, 0, force_disable_hpet},
+
 	{}
 };
 
-- 
1.9.1

